src/main/java/com/alvarium/annotators/vulnerability/GoModHandler.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators.vulnerability;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class GoModHandler implements PackageFileHandler {
+    final private File file;
+
+    protected GoModHandler(String dir) {
+        this.file = new File(dir + "/" + "go.mod");
+    }
+
+    @Override
+    public String getFileName() {
+        return this.file.getName();
+    }
+
+    @Override
+    public Map<String, String> getPackages() throws VulnerabilityException {
+        Map<String, String> packages = new HashMap<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line = null;
+            boolean inRequireBlock = false;
+            while ((line = reader.readLine()) != null) {
+                line = line.trim();
+                if (line.startsWith("require")) {
+                    inRequireBlock = true;
+                    continue;
+                }
+                if (line.startsWith(")")) {
+                    inRequireBlock = false;
+                    return packages;
+                }
+                if (inRequireBlock) {
+                    String[] parts = line.split("\\s+");
+                    if (parts.length >= 2) {
+                        String path = parts[0];
+                        String version = parts[1];
+                        packages.put(path, version);
+                    }
+                }
+            }
```

Comment:
Excerpt from the [Go Modules Reference](https://go.dev/ref/mod)

> A module is defined by a UTF-8 encoded text file named go.mod in its root directory. The go.mod file is line-oriented. Each line holds a single directive, made up of a keyword followed by arguments. For example:
>
> ```go.mod
> module example.com/my/thing
> 
> go 1.12
>
> require example.com/other/thing v1.0.2
> require example.com/new/thing/v2 v2.3.4
> exclude example.com/old/thing v1.2.3
> replace example.com/bad/thing v1.4.5 => example.com/good/thing v1.4.5
> retract [v1.9.0, v1.9.5]
>```
> The leading keyword can be factored out of adjacent lines to create a block, like in Go imports.
> ```go.mod
> require (
>    example.com/new/thing/v2 v2.3.4
>    example.com/old/thing v1.2.3
>)
>```

While it might be tricky, I think its best if we find a way to handle both of these go.mod variations and probably permutations of them as well.

Reply:
- I updated the parsing algorithm to accept:
  - Require blocks
  - Inline require
  - Permutations of both
- Written some new unit tests for these cases